lib/components/admin/mailables-window.js

```diff
@@ -0,0 +1,194 @@
+    const selectableMailables = items.filter(m =>
+      !selectedMailables.find(mailable => mailable.name === m.name)
+    )
+    return (
+      <DraggableWindow
+        header={
+          <>
+            <WindowHeader>
+              <Icon type='graduation-cap' /> Mailables{' '}
+              <span className='pull-right'>
+                <button
+                  className='clear-button-formatting'
+                  onClick={this._onClickCreateLetter}
+                  style={{marginRight: '5px', verticalAlign: 'bottom'}}
+                >
+                  Create Letter
+                </button>
+              </span>
+            </WindowHeader>
+          </>
+        }
+        onClickClose={toggleMailables}
+        scroll={false}
+        style={{height: '340px', width: '600px'}}
+      >
+        <div>
+          {MAILABLE_FIELDS.map(f => (
+            <input
+              key={f.fieldName}
+              id={f.fieldName}
+              onChange={this._updateField}
+              placeholder={f.placeholder}
+              style={{margin: '5px'}}
+              value={this.state[f.fieldName]} />
+          ))}
+        </div>
+        <div style={{display: 'flex'}}>
+          <div style={{width: '300px'}}>
+            <h4>All Mailables</h4>
+            <MailablesList>
+              {selectableMailables.map((mailable, i) => (
+                <MailableOption
+                  key={mailable.name}
+                  mailable={mailable}
+                  onClick={this._addMailable} />
+              ))}
+            </MailablesList>
+          </div>
+          <div style={{width: '300px'}}>
+            <h4>Selected Mailables <Badge>{selectedMailables.length}</Badge></h4>
+            <MailablesList>
+              {selectedMailables.length > 0
+                ? selectedMailables.map((mailable, i) => (
+                  <MailableOption
+                    index={i}
+                    key={mailable.name}
+                    mailable={mailable}
+                    onClear={this._removeMailable}
+                    updateField={this._updateMailableField} />
+                ))
+                : <div className='text-muted'>No mailables selected.</div>
+              }
+            </MailablesList>
+          </div>
+        </div>
+      </DraggableWindow>
+    )
+  }
+}
+
+class MailableOption extends Component {
+  _changeLargeFormat = (evt) => {
+    const {index, updateField} = this.props
+    updateField(index, 'largeFormat', evt.target.checked)
+  }
+
+  _changeQuantity = (evt) => {
+    const {index, updateField} = this.props
+    updateField(index, 'quantity', evt.target.value)
+  }
+
+  _onClear = () => this.props.onClear && this.props.onClear(this.props.mailable)
+
+  _onClick = () => this.props.onClick && this.props.onClick(this.props.mailable)
+
+  render () {
+    const {mailable, onClear} = this.props
+    const isSelected = Boolean(onClear)
+    if (isSelected) {
```

Comment:
I think it might be better organized to have this render method look something like:

```js
render () {
  return this.props.isSelected ? this._renderSelectedMailable() : this._renderSelectableMailable()
}
```

Reply:
Actually, scratch that, this component should actually be two components (`SelectedMailable` and (`SelectableMailable` or `MailableOption`)) so that the [Single Responsibility Principle](https://deviq.com/principles/single-responsibility-principle) can be adhered to.

Reply:
From that website:
> If you can think of more than one motivation for changing a class, it probably has more than one responsibility.

You've noted one motivation for changing the class, but if there are no others (which I don't think we can justifiably say there are), we have not really triggered the need for a new class according to that principle. I'm inclined to just keep it as one, especially after recent changes from https://github.com/opentripplanner/otp-react-redux/pull/378/commits/8ae80cb6da5e63712a54eedcd21a50ab18c5efd5

Reply:
Motivations for changing the class:

1. Not having to check if the item is selectable or not and therefore reducing the amount of code in the render method.
2. Not having to check if a handler function (`onRemove` or `onClick`) is present. If using static typing each component can be typed to require the needed handler functions. In this current state, if this were typed there'd be an additional check for whether the `updateField` property is included.
3. Removing unneeded handler functions and placing them only in components where they are needed.